cromwell_tools/_cromwell_api.py

```diff
@@ -0,0 +1,425 @@
+
+        while True:
+
+            if datetime.now() - start > timeout:
+                msg = 'Unfinished workflows after {0} minutes.'
+                raise Exception(msg.format(timeout))
+
+            all_succeeded = True
+            for uuid in workflow_ids:
+                response = cls.status(uuid, auth)
+                status = cls._parse_status(response)
+                if status in cls._failed_statuses:
+                    raise WorkflowFailedException('Workflow {0} returned status {1}'.format(
+                            uuid, status))
+                elif status != 'Succeeded':
+                    all_succeeded = False
+
+            if all_succeeded:
+                if verbose:
+                    print('All workflows succeeded!')
+                return
+
+            time.sleep(poll_interval_seconds)
+
+    @classmethod
+    def release_hold(cls, uuid, auth, raise_for_status=False):
+        """Request Cromwell to release the hold on a workflow.
+
+        It will switch the status of a workflow from ‘On Hold’ to ‘Submitted’ so it can be picked for running. For
+        a workflow that was not submitted with `workflowOnHold = true`, Cromwell will ignore the request.
+
+        Args:
+            uuid: A Cromwell workflow UUID, which is the workflow identifier. The workflow is expected to have
+                `On Hold` status.
+            auth (cromwell_tools._cromwell_auth.CromwellAuth): The authentication class holding headers or auth
+                information to a Cromwell server.
+            raise_for_status (Optional[bool]): Whether to check and raise for status based on the response. (default
+                False)
+
+        Raises:
+            requests.exceptions.HTTPError: This will be raised when raise_for_status is True and Cromwell returns
+                a response that satisfies 400 <= response.status_code < 600.
+
+        Returns:
+            response (requests.Response): HTTP response from Cromwell.
+        """
+        response = cls.requests.post(url=auth.url + cls._release_hold_endpoint.format(uuid=uuid),
+                                     auth=auth.auth,
+                                     headers=auth.header)
+        if raise_for_status:
+            cls._check_and_raise_status(response)
+        return response
+
+    @classmethod
+    def query(cls, query_dict, auth, raise_for_status=False):
+        """
+        TODO: Given that Cromwell-as-a-Service blocks a set of features that are available in Cromwell, e.g. 'labelor',
+        for security concerns, the first iteration of this API doesn't come up with the advanced query keys of the
+        Cromwell except a set of necessary ones. However, we need to implement this for completeness and keep an eye
+        on the compatibility between CaaS and Cromwell.
+
+        _All of the query keys will be used in an OR manner, except the keys within `labels`, which are defined in
```

Comment:
Why are there underscores around this sentence?

Reply:
🤔 IDK... Let me remove them!